internal/service/substrate.go

```diff
@@ -106,15 +106,30 @@ func max(a, b int) int {
 	return b
 }
 
+type BlockFinalized struct {
+	BlockNum  int  `json:"block_num"`
+	Finalized bool `json:"finalized"`
+}
+
+func (s *SubscribeService) catchUp(pool *ants.PoolWithFunc, wg *sync.WaitGroup) {
+	missing := s.dao.GetMissingBlockNums()
+
+	for _, blockNum := range missing {
+		slog.Info("catchUp", "blockNum", blockNum)
+		wg.Add(1)
+		err := pool.Invoke(BlockFinalized{BlockNum: blockNum, Finalized: true})
+		if err != nil {
+			logError("ChainGetBlockHash get", err)
+		}
+	}
+	wg.Wait()
+}
+
 func (s *SubscribeService) subscribeFetchBlock() {
 	var wg sync.WaitGroup
 	ctx := context.TODO()
-	type BlockFinalized struct {
-		BlockNum  int  `json:"block_num"`
-		Finalized bool `json:"finalized"`
-	}
 
-	p, _ := ants.NewPoolWithFunc(50, func(i interface{}) {
+	p, _ := ants.NewPoolWithFunc(10, func(i interface{}) {
```

Comment:
Do you see improvements with 10?

Reply:
So I started getting these "can't assign address errors" when trying to talk to the DB:

```
2023/06/20 19:50:34 /home/subscan-essentials/internal/dao/chainExtrinsic.go:70 dial tcp 127.0.0.1:3306: connect: can't assign requested address
```
My panicky, rushed thought process was something along the lines of

maybe it's too many open connections -> shouldn't this be handled by gorm's pooling -> well let's just try reducing the number of worker threads anyway -> it works -> wtf why does it work -> do i really have time to debug it

since it fixed it i left it in, but seems like it's a bit of a yellow/orange flag